tests/unit/test-cmd/test.sign.js

```diff
@@ -278,8 +278,10 @@ describe('sign', () => {
                        stubs.signingConfig.timeout);
           assert.equal(signedAddonCall.xpiPath,
                        stubs.buildResult.extensionPath);
-          assert.equal(signedAddonCall.id,
-                       stubs.preValidatedManifest.applications.gecko.id);
+
+          const applications = stubs.preValidatedManifest.applications || {};
```

Comment:
Huh, I didn't think about that. Makes sense, I changed it to to what you suggested. This makes the defensive code quite annoying though and I've run into this elsewhere. I think it's time to move all manifest access to an object that has a method like `manifest.getGeckoId()`: https://github.com/mozilla/web-ext/issues/641